src/rebar_paths.erl

```diff
@@ -208,4 +212,27 @@ get_apps(deps, State) ->
     end ++
     rebar_state:all_deps(State);
 get_apps(plugins, State) ->
-    rebar_state:all_plugin_deps(State).
+    rebar_state:all_plugin_deps(State);
+get_apps(runtime, State) ->
+    %% We get all project apps and for each of them we find
+    %% their runtime deps (i.e., `applications' and `included_applications').
+    ProjectApps = rebar_state:project_apps(State),
+    RuntimeApps =
+        [begin
+            Apps = rebar_app_info:applications(App),
+            IncludedApps = rebar_app_info:included_applications(App),
+            lists:foldl(
+              fun(AppName0, Acc) ->
+                  %% If the App is an atom convert it to a binary, otherwise leave it as it is.
+                  AppName1 = if is_atom(AppName0) -> atom_to_binary(AppName0, utf8);
+                                true -> AppName0
+                             end,
+                  %% We only care about those apps we could find in the `State'.
+                  case rebar_app_utils:find(AppName1, ProjectApps) of
+                      {ok, AppInfo} -> [AppInfo|Acc];
+                      error -> Acc
+                  end
+              end, [App], Apps ++ IncludedApps)
+         end || App <- ProjectApps],
+    %% We get rid of duplicated apps from the flattened list.
+    lists:usort(lists:flatten(RuntimeApps)).
```

Comment:
I think this code is going to be tricky and currently may not work because it needs to be done recursively.

For example, if I have `A` include `B` and `C`, and that `C` includes `D`, I need all of `A`, `B`, `C`, and `D` to be in path because that's how something like relx or `application:ensure_all_started` will resolve them.

This call therefore need to be recursive, and possibly build a working set of all transitive apps that could be found by recursing from the project apps.

Reply:
You're totally right, while testing I didn't notice I was missing some deps! :facepalm: